### PR TITLE
Add consumers per channel limit

### PR DESCRIPTION
**Pull request by illotum**
## Proposed Changes

This adds a simple per channel limit to the number of consumers. Addresses the raised recently issue with consumer leaks leading to crashes.

It does affect "reply-to" consumers. But doesn't run in `rabbit_channel:handle_consuming_queue_down_or_eol`. Open to suggestions regarding these cases.

Note that crossing the limit crashes the connection. Same behavior as channel limit, so I figured it's acceptable.

Additionally, fixes a bit of docs and tests for `channel_max_per_node`. 

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes issue #NNNN)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [x] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)
- [ ] Build system and/or CI

## Checklist

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All tests pass locally with my changes
- [ ] If relevant, I have added necessary documentation to https://github.com/rabbitmq/rabbitmq-website
- [ ] If relevant, I have added this change to the first version(s) in release-notes that I expect to introduce it

### CODE REVIEW

File: deps/rabbit/src/rabbit_channel.erl [hunk 1]
```diff
@@ -1313,8 +1316,13 @@ handle_method(#'basic.consume'{queue        = <<"amq.rabbitmq.reply-to">>,
                                no_ack       = NoAck,
                                nowait       = NoWait},
               _, State = #ch{reply_consumer   = ReplyConsumer,
+                             cfg = #conf{max_consumers = MaxConsumers},
                              consumer_mapping = ConsumerMapping}) ->
+    CurrentConsumers = length(maps:keys(ConsumerMapping)),
```
Comment (SimonUnge) [inline deps/rabbit/src/rabbit_channel.erl:1321]: maps:size?

File: deps/rabbit/priv/schema/rabbit.schema [hunk 2]
```diff
@@ -889,6 +889,11 @@ end}.
 
 {mapping, "channel_max", "rabbit.channel_max", [{datatype, integer}]}.
 
+%% Set the max allowed number of channels per node.
+%% `infinity` means "no limit".
+%%
+%% {channel_max_per_node, infinity},
```
Comment (SimonUnge) [inline deps/rabbit/priv/schema/rabbit.schema:895]: A good comment, but as it is unrelated to this PR, I would avoid adding it.

File: deps/rabbit/src/rabbit_channel.erl [hunk 3]
```diff
@@ -106,6 +106,7 @@
           consumer_timeout,
           authz_context,
           %% defines how ofter gc will be executed
```
Comment (michaelklishin) [inline deps/rabbit/src/rabbit_channel.erl:108]: This line is referring to `writer_gc_threshold`, let's move it around accordingly and keep all comments on the previous line.

**Comment by michaelklishin** [comment]
Merged with a couple of minor changes in https://github.com/rabbitmq/rabbitmq-server/pull/10754.

**Pull request closed without merging**